### DISCUSSION TITLE
How to use assert_abs_diff_eq with an arr2?

### DISCUSSION THREAD

**Discussion by neilpate** [rust-ndarray/ndarray#1239] [Q&A]
Sorry I am terrible at formatting markdown so I pasted a picture of the code as well.
I have a test that I want to use to check some 2D array maths. I want to do an assert on an approximately equal value. The last line does not work as the trait is not implemented.

![image](https://user-images.githubusercontent.com/7802334/205518569-ac83bbf8-7a15-4bca-910e-4cd83720be10.png)

![image](https://user-images.githubusercontent.com/7802334/205518546-a8b066af-22c5-4f36-8372-41dc14e23ef2.png)

#[test]
fn test_create_y_rotation_matrix() {
    let expected = arr2(&[
        [0.93969262, 0., 0.34202015, 0.],
        [0., 1., 0., 0.],
        [-0.34202015, 0., 0.93969262, 0.],
        [0., 0., 0., 1.],
    ]);

    let result = Create_Y_Rotation_Matrix(20.);

    assert_abs_diff_eq!(expected, result);                                                                                            
}

**Comment by bluss** [comment]
Your inline type hints suggest one array has elements of f64 and the other f32, otherwise they should be the same type. I'd check aligning them on type first.

**Comment by neilpate** [comment]
Thanks, I forced to f32 for both but it still gives the same error

![image](https://user-images.githubusercontent.com/7802334/206872766-98025bb2-5913-4dd7-bf64-5d7d9554ce56.png)

![image](https://user-images.githubusercontent.com/7802334/206872776-83378c50-5fc4-44c9-96e0-b15b34dc3f82.png)
